mula/scheduler/models/tasks.py

```diff
@@ -57,10 +59,58 @@ class Task(BaseModel):
 
     status: TaskStatus
 
+    # Durations
+    pending: Optional[timedelta] = None
+    queued: Optional[timedelta] = None
+    dispatched: Optional[timedelta] = None
+    running: Optional[timedelta] = None
```

Comment:
Added new duration columns for setting specifically or by calculation

Reply:
Perhaps it makes more sense to save datetimes of when the state was (last) changed? Then these fields could be called `[...]_at` so the naming would be clearer.

Also, when a task gets dispatched but re-queued because of an error or shutdown, this route would potentially bring a lot of complex calculations with it to get the right numbers. I am btw in general always a bit hesitant about working with interval-like data structures vs. static data points.

Reply:
Updated, made it timestamps. Now if we want to make queries: "What tasks were queued longer than 5 minutes in the past 6 days" we need to write specific queries for it that calculate the difference between those timestamps